src/plugins/data/public/ui/search_bar/search_bar.tsx

```diff
@@ -467,6 +474,7 @@ class SearchBarUI extends Component<SearchBarProps, State> {
             useSaveQueryMenu={useSaveQueryMenu}
             isQueryEditorControl={isQueryEditorControl}
             saveQuery={this.onSave}
+            queryStringManager={this.props.queryStringManager}
```

Comment:
the query string manager being pass in which will impact the state of the components in search bar which is also utilized by other plugins besides the Discover plugin (like dashboards and visualizations) so i would watch out and generally avoid touching this file.

this already was able to access the query string manager like this:

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/public/ui/search_bar/search_bar.tsx#L123

this should be available in similar locations within this ui

Reply:
Thanks for calling that out. I'll revert line 205 in [create_search_bar.tsx](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8848/files#diff-c9b47a56764f7a94d072874feb9b1f3473dedb7d32480ace6405475c3e953ce6).

Reply:
Addressed in https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8864